src/Filter/FilterInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Everlution\Navigation\Filter;
+
+use Everlution\Navigation\Item;
+
+/**
+ * @author Klinec <matus.pavlik@everlution.com>
+ */
+interface FilterInterface
+{
+    /**
+     * @param Item $item
+     * @return bool
+     */
+    public function isApplicable(Item $item): bool;
+
+    /**
+     * @param Item $item
+     * @return bool
+     */
+    public function filterOut(Item $item): bool;
```

Comment:
This should be named more explicitly - it returns bool so it should be named like `is*`, `has*`, `should*` ...

`filterOut()` indicates action but we are just "testing" the state and upon this test we are going to do some action (filter out or not). The most appropriate would probably be `shouldFilterOut()`